components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/utils/ExportUtils.java

```diff
@@ -0,0 +1,1087 @@
+                }
+            } else if (log.isDebugEnabled()) {
+                log.debug("Thumbnail URL [" + thumbnailUrl + "] does not exists in registry for API/API Product: "
+                        + identifier.getName() + StringUtils.SPACE + APIConstants.API_DATA_VERSION + ": "
+                        + identifier.getVersion() + ". Skipping thumbnail export.");
+            }
+        } catch (RegistryException e) {
+            log.error("Error while retrieving API/API Product Thumbnail " + thumbnailUrl, e);
+        } catch (IOException e) {
+            //Exception is ignored by logging due to the reason that Thumbnail is not essential for
+            //an API to be recreated.
+            log.error("I/O error while writing API/API Product Thumbnail: " + thumbnailUrl + " to file", e);
+        }
+    }
+
+    /**
+     * Retrieve SOAP to REST mediation logic for the exporting API and store it in the archive directory
+     *
+     * @param archivePath   File path to export the SOAPToREST mediation logic
+     * @param apiIdentifier ID of the requesting API
+     * @param registry      Current tenant registry
+     * @throws APIImportExportException If an error occurs while retrieving image from the registry or
+     *                                  storing in the archive directory
+     */
+    public static void addSOAPToRESTMediationToArchive(String archivePath, APIIdentifier apiIdentifier,
+                                                       UserRegistry registry) throws APIImportExportException {
+        String soapToRestBaseUrl = "/apimgt/applicationdata/provider" + RegistryConstants.PATH_SEPARATOR +
+                apiIdentifier.getProviderName() + RegistryConstants.PATH_SEPARATOR +
+                apiIdentifier.getApiName() + RegistryConstants.PATH_SEPARATOR +
+                apiIdentifier.getVersion() + RegistryConstants.PATH_SEPARATOR +
+                SOAPToRESTConstants.SOAP_TO_REST_RESOURCE;
+
+        InputStream inputStream = null;
+        OutputStream outputStream = null;
+        try {
+            if (registry.resourceExists(soapToRestBaseUrl)) {
+                Collection inFlow = (org.wso2.carbon.registry.api.Collection) registry.get(soapToRestBaseUrl
+                        + RegistryConstants.PATH_SEPARATOR + IN);
+                Collection outFlow = (org.wso2.carbon.registry.api.Collection) registry.get(soapToRestBaseUrl
+                        + RegistryConstants.PATH_SEPARATOR + OUT);
+
+                CommonUtil.createDirectory(archivePath + File.separator + SOAPTOREST + File.separator + IN);
+                CommonUtil.createDirectory(archivePath + File.separator + SOAPTOREST + File.separator + OUT);
+                if (inFlow != null) {
+                    for (String inFlowPath : inFlow.getChildren()) {
+                        inputStream = registry.get(inFlowPath).getContentStream();
```

Comment:
Any reason for not using `try with resources` pattern here, we can avoid needing to close the stream.

Reply:
Fixed via https://github.com/wso2/carbon-apimgt/pull/9406/commits/150d9abd9d4eb25c7f1217fc53f6143f94167542